### ISSUE TITLE
Smaller fraction subtracted by bigger fraction wrong output

### ISSUE THREAD

**Issue by aarontxz** [nus-cs2030/2223-s2#804]
basically when my code runs 
Fraction.of(1, 4).sub(Fraction.of(1, 2))

it outputs
NaN : 8

instead of 
NaN

anyone have any suggestions on how to handle this?

**Comment by syafiq9326** [comment]
#759

think ur facing the same issue as I did, you can refer to this post, I ended up using a filter function on my optional<Number> like what was suggested. 

filter(x -> x.isValid())

hope this helps!

**Comment by WXUXIN** [comment]
@syafiq9326 is right, the logic would be to check the validity of both the resultant numerator and denominator

**Comment by ryantangmj** [comment]
You can map the numerator and denomator to isValid() then compare it to Optional.of(true) to check if the numerator and denominator are valid

**Comment by meljwang** [comment]
Hi, you can check the validity of both the resultant numerator and denominator and return a new Fraction with an Optional.empty if either the numerator or denominator is invalid.

**Comment by e0969533** [comment]
yep you should add a condition that checks the numerator/denominator by isValid() and using Optional.of(true), if it is valid, return the fraction, else return an optional empty

**Comment by Aron113** [comment]
use the filter predicate and then isValid()

**Comment by leonardwx** [comment]
yeap i just resolved the issue using filter(x -> x.isValid())

**Comment by aarontxz** [comment]
Thanks, everyone using filter worked! I used the filter method using isvalid as the predicate, then I check if after the filter the numerator is now equaled to optional empty if so I return new fraction optional empty in my sub method.

**Comment by meganyeoyy** [comment]
I had the same issue too, but after using filter method it worked.
